smart-client-jersey/src/main/java/com/emc/rest/smart/jersey/SmartClientFactory.java

```diff
@@ -142,8 +146,10 @@ static ApacheHttpClient4Handler createApacheClientHandler(SmartConfig smartConfi
         //       PoolingHttpClientConnectionManager will break threading)
         org.apache.http.impl.conn.PoolingClientConnectionManager connectionManager = new org.apache.http.impl.conn.PoolingClientConnectionManager();
         // 999 maximum active connections (max allowed)
-        connectionManager.setDefaultMaxPerRoute(999);
-        connectionManager.setMaxTotal(999);
+        Object maxConnectionsPerHost = smartConfig.getProperty(MAX_CONNECTIONS_PER_HOST);
+        Object maxConnections = smartConfig.getProperty(MAX_CONNECTIONS);
+        connectionManager.setDefaultMaxPerRoute(smartConfig.getPropertyIntValue(maxConnectionsPerHost, MAX_CONNECTIONS_PER_HOST_DEFAULT));
```

Comment:
with proposed changes in `SmartConfig`, change this to `smartConfig.getIntProperty(MAX_CONNECTIONS_PER_HOST, MAX_CONNECTIONS_PER_HOST_DEFAULT)`

... similar for the following line.